### PR TITLE
LPS-183396 Enable account restriction for Proxy Salesforce object

### PR DESCRIPTION
**Pull request by nNilton**

### CODE REVIEW

**Comment by guilhermedcamacho** [comment]
Just started reviewing :)

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 1]
```diff
@@ -1810,11 +1812,12 @@ private void _assertObjectDefinitions2() throws Exception {
 			_objectDefinitionLocalService.fetchObjectDefinition(
 				_group.getCompanyId(), "C_TestObjectDefinition4");
 
+		Assert.assertTrue(objectDefinition4.isAccountEntryRestricted());
 		Assert.assertTrue(
 			objectDefinition4.getAccountEntryRestrictedObjectFieldId() > 0);
-		Assert.assertTrue(objectDefinition4.isAccountEntryRestricted());
 		Assert.assertTrue(objectDefinition4.isEnableCategorization());
 		Assert.assertTrue(objectDefinition4.isEnableComments());
+		Assert.assertEquals("default", objectDefinition4.getStorageType());
```
Comment (guilhermedcamacho) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:1820]: Please use the constant `ObjectDefinitionConstants#STORAGE_TYPE_DEFAULT` or method `ObjectDefinition#isDefaultStorageType`

I prefer this pattern `Assert.assertTrue(objectDefinition4.isDefaultStorageType());`

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 2]
```diff
@@ -1275,18 +1280,27 @@ private Map<String, String> _addObjectDefinitions(
...
+				_objectFieldLocalService.fetchObjectField(
+					serviceBuilderObjectDefinition.getObjectDefinitionId(),
+					entry.getValue(
+					).getAccountEntryRestrictedObjectFieldName());
 
-		_invoke(
-			() -> _addOrUpdateObjectFields(
-				listTypeDefinitionIdsStringUtilReplaceValues,
-				objectDefinitionIdsStringUtilReplaceValues, serviceContext));
+			if (Objects.equals(
```
Comment (guilhermedcamacho) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:1289]: What do you think to use the method `ObjectDefinition#isDefaultStorageType`?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]
Hey @nNilton , I left some comments. Let me know if you have questions. Thanks!

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java [hunk 3]
```diff
@@ -1111,6 +1113,122 @@ public void testEnableAccountEntryRestricted() throws Exception {
...
+		Assert.assertTrue(
+			objectDefinition1.getAccountEntryRestrictedObjectFieldId() > 0);
+		Assert.assertFalse(objectDefinition1.isSystem());
+		Assert.assertEquals(
+			objectDefinition1.getStorageType(),
+			ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE);
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
+
+		// Enabling account restriction for default storage object definition
```
[the remark this answers is not in the document]
Comment (nNilton) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java:1160]: This is my goal, but I'll discuss this with bchan

**Comment by nNilton** [comment]
@guilhermedcamacho could you please take a look at this once again? Thanks in advance

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java [hunk 4]
```diff
@@ -1111,6 +1113,126 @@ public void testEnableAccountEntryRestricted() throws Exception {
...
+		// using salesforce method
+
+		User user = TestPropsValues.getUser();
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java:1163]: Can you inline this when invoking `_objectDefinitionLocalService.addObjectDefinition`?
I think you can use TestPropsValues.getUserId() directly

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 5]
```diff
@@ -527,6 +527,37 @@ public ObjectDefinition enableAccountEntryRestricted(
...
+			throw new ObjectDefinitionAccountEntryRestrictedException(
+				"Custom object definitions can only be restricted by a " +
+					"Integer, Long Integer or Text field");
+		}
+
+		ObjectDefinition objectDefinition = getObjectDefinition(
+			objectField.getObjectDefinitionId());
+
+		if (objectDefinition.isDefaultStorageType() ||
+			objectDefinition.isAccountEntryRestricted()) {
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:548]: How to change the accountEntryRestrictedObjectFieldId for an objectDefinition preconfigured as accountEntryRestricted?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Review by guilhermedcamacho — APPROVED** [review]
LGTM! :tada:

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
Hey @nNilton , can you fix semantic versioning problem for `com.liferay.object.api` module? Thanks!

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
Hey @nNilton , can you fix the conflicting files? Thanks!

**Comment by guilhermedcamacho** [comment]
ci:forward

**Pull request closed without merging**